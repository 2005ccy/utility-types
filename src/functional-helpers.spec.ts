import { testType } from './test-utils';
import { getReturnOfExpression } from '.';

//#region Docs Example
const increment = () => ({ type: 'INCREMENT' as 'INCREMENT' });

const returnOfIncrement = getReturnOfExpression(increment);
type IncrementAction = typeof returnOfIncrement; // { type: "INCREMENT"; }
//#endregion

describe('Type Utils', () => {
  describe('$call', () => {
    it('should return null value', () => {
      expect(returnOfIncrement).toBe(undefined);
      testType<{ type: 'INCREMENT' }>(returnOfIncrement);
    });
  });
});
