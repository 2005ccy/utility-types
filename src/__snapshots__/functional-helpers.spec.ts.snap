// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getReturnOfExpression getReturnOfExpression(increment) (type) should match snapshot 1`] = `"{ type: \\"INCREMENT\\"; }"`;
